File: app-presentation/src/main/java/com/zxjdev/smile/presentation/moment/MomentModelMapper.java
package com.zxjdev.smile.presentation.moment;

import com.zxjdev.smile.domain.base.BaseMapper;
import com.zxjdev.smile.domain.moment.Moment;
import com.zxjdev.smile.presentation.user.UserModelMapper;

import javax.inject.Inject;

public class MomentModelMapper extends BaseMapper<MomentModel, Moment> {

  @Inject UserModelMapper userModelMapper;

  @Inject
  public MomentModelMapper() {

  }

  @Override
  public MomentModel transform(Moment data) {
    MomentModel model = new MomentModel();
    model.setOwner(userModelMapper.transform(data.getOwner()));
    model.setContent(data.getContent());
    model.setCreateAt(data.getCreateAt());
    return model;
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/moment/MomentModelMapper.kt
package com.zxjdev.smile.presentation.moment

import com.zxjdev.smile.domain.common.base.BaseMapper
import com.zxjdev.smile.domain.moment.Moment
import com.zxjdev.smile.presentation.user.UserModelMapper

import javax.inject.Inject

class MomentModelMapper @Inject
constructor() : BaseMapper<MomentModel, Moment>() {

    @Inject lateinit var userModelMapper: UserModelMapper

    override fun transform(data: Moment): MomentModel {
        val model = MomentModel()
        model.owner = userModelMapper!!.transform(data.owner)
        model.content = data.content
        model.createAt = data.createAt
        return model
    }
}
